Objects/longobject.c

```diff
@@ -6681,3 +6681,84 @@ Py_ssize_t
 PyUnstable_Long_CompactValue(const PyLongObject* op) {
     return _PyLong_CompactValue(op);
 }
+
+PyObject* PyLong_FromInt32(int32_t value)
+{ return PyLong_FromNativeBytes(&value, sizeof(value), -1); }
+
+PyObject* PyLong_FromUInt32(uint32_t value)
+{ return PyLong_FromUnsignedNativeBytes(&value, sizeof(value), -1); }
+
+PyObject* PyLong_FromInt64(int64_t value)
+{ return PyLong_FromNativeBytes(&value, sizeof(value), -1); }
+
+PyObject* PyLong_FromUInt64(uint64_t value)
+{ return PyLong_FromUnsignedNativeBytes(&value, sizeof(value), -1); }
+
+int PyLong_ToInt32(PyObject *obj, int32_t *value)
+{
+#if SIZEOF_INT == 4
+    int res = PyLong_AsInt(obj);
```

Comment:
Why not just use `PyLong_AsNativeBytes` for each of these?

Reply:
It's just old habits :-) I updated the implementation of "To" functions to use `PyLong_AsNativeBytes()`. It has two side effects:

* Use `__index__()` method if available.
* Always raise `ValueError` if the value is negative.

@serhiy-storchaka: Using `__index__()` is a good thing, no? :-)

Reply:
> Using `__index__()` is a good thing, no? :-)

If it's good enough for `PyLong_AsLong` then I don't see why it isn't good enough for these functions.

Reply:
`PyLong_AsLong(obj)` calls `obj.__index__()`, whereas `PyLong_AsUnsignedLong(obj)` doesn't call `obj.__index__()`. I don't know the history of the `PyLong_AsUnsignedLong()` design, if there was a "design" for this function :-)

For new functions, I propose to call `obj.__index__()`.

Reply:
The code now uses PyLong_AsNativeBytes().